compiler/ilgen/OMRJitBuilderRecorder.hpp

```diff
@@ -0,0 +1,212 @@
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_JITBUILDERRECORDER_INCL
+#define OMR_JITBUILDERRECORDER_INCL
+
+#include <iostream>
+#include <fstream>
+#include <map>
+
+namespace TR { class IlBuilderRecorder; }
+namespace TR { class MethodBuilderRecorder; }
+namespace TR { class IlType; }
+namespace TR { class IlValue; }
+
+namespace OMR
+{
+
+class JitBuilderRecorder
+   {
+   public:
+   const int16_t VERSION_MAJOR=0;
+   const int16_t VERSION_MINOR=0;
+   const int16_t VERSION_PATCH=0;
+   const char *RECORDER_SIGNATURE = "JBIL";
+   const char *JBIL_COMPLETE      = "Done";
+
+   typedef uint32_t                      TypeID;
+   typedef std::map<const void *,TypeID> TypeMapID;
+
+   JitBuilderRecorder(const TR::MethodBuilderRecorder *mb);
+   virtual ~JitBuilderRecorder();
+
+   void setMethodBuilderRecorder(TR::MethodBuilderRecorder *mb) {_mb = mb;}
+
+   /**
+    * @brief Subclasses override these functions to record to different output formats
+    */
+   virtual void Close()                                       { }
+   virtual void String(const char * const string)             { }
+   virtual void Number(int8_t num)                            { }
+   virtual void Number(int16_t num)                           { }
+   virtual void Number(int32_t num)                           { }
+   virtual void Number(int64_t num)                           { }
+   virtual void Number(float num)                             { }
+   virtual void Number(double num)                            { }
+   virtual void ID(TypeID id)                                 { }
+   virtual void Statement(const char *s)                      { }
+   virtual void Type(const TR::IlType *type)                  { }
+   virtual void Value(const TR::IlValue *v)                   { }
+   virtual void Builder(const TR::IlBuilderRecorder *b)       { }
+   virtual void Location(const void * location)               { }
+
+   virtual void BeginStatement(const TR::IlBuilderRecorder *b, const char *s);
+   virtual void BeginStatement(const char *s);
+   virtual void EndStatement()                                { }
+
+   void StoreID(const void *ptr);
+   bool EnsureAvailableID(const void *ptr);
+
+   /**
+    * @brief constant strings used to define output format
+    */
+   const char *STATEMENT_DEFINENAME                   = "DefineName";
```

Comment:
To fix the Windows compile failure all of these fields need to be static.
`static const char *STATEMENT_DEFINENAME                   = "DefineName";`

but to be even more pedantic I think it makes more sense to declare them as const pointers to const data.
`static const char *const STATEMENT_DEFINENAME                   = "DefineName";`